src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-90400.mdx

```diff
@@ -0,0 +1,48 @@
+---
+subject: Python agent
+releaseDate: '2024-01-08'
+version: 9.4.0
+downloadLink: 'https://pypi.python.org/pypi/newrelic'
+features: ['Add nonced context security policy support', 'Add wheels for Python 3.12 and musl', 'Publish newrelic.agent.CallableObjectProxy API', 'Deprecate ObjectWrapper API', 'Deprecate get_browser_timing_footer API']
+bugs: ['Fix distributed tracing headers in kafka', 'Fix case sensitive header search in ASGIBrowserMiddleware']
+security: []
+---
+
+## Notes
+
+This release of the Python agent adds support for nonced content security policies for Browser Agent injection, adds new wheels for Python 3.12 and musl Linux systems such as Alpine, deprecates the ObjectWrapper and `get_browser_timing_footer` APIs, fixes a crash with Kafka instrumentation when using distributed tracing, and removes a case sensitive header check in `ASGIBrowserMiddleware`.
+
+Install the agent using `easy_install/pip/distribute` via the [Python Package Index](https://pypi.python.org/pypi/newrelic) or download it directly from the [New Relic download site](https://download.newrelic.com/python_agent/release).
+
+## Deprecations
+
+* **Deprecate `newrelic.agent.ObjectWrapper` API**
+Removes custom `newrelic.agent.ObjectWrapper` implementation and instead creates a pass through to `wrapt.FunctionWrapper`.
+
+* **Deprecate `get_browser_timing_footer` API**
+Previously `get_browser_timing_footer` was an API that would be used to supply information to the Browser Agent loaded by `get_browser_timing_header`. This step is now included in `get_browser_timing_header` making `get_browser_timing_footer` obsolete. This function now returns an empty string to avoid breaking existing applications.
```

Comment:
```suggestion
* Deprecated `newrelic.agent.ObjectWrapper` API
    * Removes custom `newrelic.agent.ObjectWrapper` implementation and instead creates a pass through to `wrapt.FunctionWrapper`.

* Deprecated `get_browser_timing_footer` API
    * Previously `get_browser_timing_footer` was an API that would be used to supply information to the browser agent loaded by `get_browser_timing_header`. 
    * This step is now included in `get_browser_timing_header`, making `get_browser_timing_footer` obsolete. This function now returns an empty string to avoid breaking existing applications.
```